### PR TITLE
SDKS-1009 Expose API to get all push notifications stored

### PR DESCRIPTION
**Pull request by rodrigoareis**
On this change a new getAllNotifications() method is exposed on the FRAClient class. This will allow developers to retrieve a single list of notifications across all mechanisms.

### CODE REVIEW

File: FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift [hunk 1]
```diff
@@ -800,7 +798,160 @@ class FRAClientTests: FRABaseTests {
...
+        aps["sound"] = "default"
+
+        payload["aps"] = aps
+
+        guard let _ = FRAPushHandler.shared.application(UIApplication.shared, didReceiveRemoteNotification: payload) else {
+          XCTFail("Failed to parse notification payload and construct PushNotification object")
+          return
+        }
+
+        var pushAccount = authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser")
```
Comment (jhoongo) [inline FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift:886]: maybe reuse the Account instance retrieved from line#852?

File: FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift [hunk 2]
```diff
@@ -800,7 +798,160 @@ class FRAClientTests: FRABaseTests {
...
+
+        payload["aps"] = aps
+
+        guard let _ = FRAPushHandler.shared.application(UIApplication.shared, didReceiveRemoteNotification: payload) else {
+          XCTFail("Failed to parse notification payload and construct PushNotification object")
+          return
+        }
+
+
+        pushAccount = authenticatorManager.getAccount(identifier: "ForgeRockSandbox-pushtestuser")
```
Comment (jhoongo) [inline FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift:911]: Not quite sure on this, perhaps I might forget on something, is there any reason that we keep reading the object from the storage?

File: FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift (hunk 1 above)
Comment (jhoongo) [reply FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift:886]: Hmm.. I guess that's what I was doing in other tests.. it's may bad.

**Review by jhoongo — DISMISSED** [review]
Nothing major, looks good to me.

File: FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift (hunk 1 above)
Comment (rodrigoareis) [reply FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift:886]: Yes, I reused the previous```test_07_get_all_notifications_for_mechanism``` as a baseline. The reason it's needed is the following Assert:

`XCTAssertEqual(pushMechanism?.notifications.count, 1)`

The ```notifications``` attribute under PushMechanism is only initialized when an account is retrieved using `getAccount()` or `getAllAccounts()`.

File: FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift (hunk 2 above)
Comment (rodrigoareis) [reply FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift:911]: Same as above.

File: FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift (hunk 1 above)
Comment (jhoongo) [reply FRAuthenticator/FRAuthenticatorTests/E2ETests/FRAClient/FRAClientTests.swift:886]: I see, that makes sense. Perhaps it's something that we may want to discuss further on how/when we should read it from the storage, but for this story itself, I believe that it should be okay.

Thanks,

**Comment by jhoongo** [comment]
@rodrigoareis Can you address the CHANGELOG.md conflict? As I merged the other PR, I think it created the conflict.

**Review by george-bafaloukas-forgerock — DISMISSED** [review]

**Review by spetrov — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**